core/ulog/log.go

```diff
@@ -60,21 +60,45 @@ type baselogger struct {
 type Log interface {
 	// Configure Log object with the provided log.Configuration
 	Configure(Configuration)
```

Comment:
It's fairly natural to do something like this
```go
log := ulog.LogBuilder().WithConfig(&Configuration{}).SetLevel(..).Build()
```

Once the logger is built, it's best if it can be immutable w.r.t. configuration parameters.